src/Interpreters/executeQuery.cpp

```diff
@@ -378,6 +386,39 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
             /// TODO: parser should fail early when max_query_size limit is reached.
             ast = parseQuery(parser, begin, end, "", max_query_size, settings.max_parser_depth);
         }
+        #if USE_PRQL
+        else if (settings.dialect == Dialect::prql && !internal)
+        {
+            char * prql_query = new char[max_query_size + 1];
+            if (prql_query == nullptr)
+            {
+                throw Exception(ErrorCodes::CANNOT_ALLOCATE_MEMORY, "Not enough RAM");
+            }
```

Comment:
`new` will throw `std::bad_alloc`, this is redundant

Reply:
This part is heavily refactored, check https://github.com/ClickHouse/ClickHouse/blob/6df1fa0815bc9bb34bb8f57419520d2d5f6ba550/src/Parsers/PRQL/ParserPRQLQuery.cpp#L25-L82